### PR TITLE
[VL] Make the maxBatchSize check consistent

### PR DESCRIPTION
**Pull request by WangGuangxin**
## What changes were proposed in this pull request?
A minor change to make the `maxBatchSize` consistent with the check in `VeloxShuffleWriter` 

https://github.com/apache/incubator-gluten/blob/main/cpp/velox/shuffle/VeloxShuffleWriter.cc#L222

### CODE REVIEW

**Comment by WangGuangxin** [comment]
cc @marin-ma

**Comment by marin-ma** [comment]
@FelixYBW Should we completely remove the batch size limitation on both java and native code as https://github.com/apache/incubator-gluten/pull/5326 has been merged?

**Comment by FelixYBW** [comment]
> @FelixYBW Should we completely remove the batch size limitation on both java and native code as #5326 has been merged?

Yes, let's comment it and marked as issue to be solved in Velox.

File: backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala [hunk 1]
```diff
@@ -481,10 +481,10 @@ object BackendSettings extends BackendSettingsApi {
   private def checkMaxBatchSize(nativeConf: java.util.Map[String, String]): Unit = {
     if (nativeConf.containsKey(GlutenConfig.GLUTEN_MAX_BATCH_SIZE_KEY)) {
       val maxBatchSize = nativeConf.get(GlutenConfig.GLUTEN_MAX_BATCH_SIZE_KEY).toInt
-      if (maxBatchSize > MAXIMUM_BATCH_SIZE) {
+      if (maxBatchSize >= MAXIMUM_BATCH_SIZE) {
```
Comment (marin-ma) [inline backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala:484]: The check in native shuffle writer requires the split buffer (aka. output batch size from shuffle writer) <= 32k, which means the maximum batch size in the pipeline can equal to 32k. But this change makes the maximum batch size < 32k.

Comment (WangGuangxin) [reply backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala:484]: Got it

Comment (FelixYBW) [reply backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala:484]: @marin-ma we should remove the check here, right?

**Pull request closed without merging**